cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -254,25 +257,27 @@ public KafkaBrokerConfigurationBuilder withKRaft(String clusterName, String name
      * generate the per-broker configuration which uses actual broker IDs and addresses instead of just placeholders.
      *
      * @param clusterName                Name of the cluster (important for the advertised hostnames)
+     * @param kafkaVersion
      * @param namespace                  Namespace (important for generating the advertised hostname)
      * @param kafkaListeners             The listeners configuration from the Kafka CR
      * @param advertisedHostnameProvider Lambda method which provides the advertised hostname for given listener and
      *                                   broker. This is used to configure the user-configurable listeners.
      * @param advertisedPortProvider     Lambda method which provides the advertised port for given listener and broker.
      *                                   This is used to configure the user-configurable listeners.
-     *
      * @return Returns the builder instance
      */
     @SuppressWarnings({"checkstyle:CyclomaticComplexity"})
     public KafkaBrokerConfigurationBuilder withListeners(
             String clusterName,
+            KafkaVersion kafkaVersion,
             String namespace,
             List<GenericKafkaListener> kafkaListeners,
             Function<String, String> advertisedHostnameProvider,
             Function<String, String> advertisedPortProvider
     )  {
         List<String> listeners = new ArrayList<>();
-        List<String> advertisedListeners = new ArrayList<>();
+        List<String> brokerAdvertisedListeners = new ArrayList<>();
+        List<String> controllerAdvertisedListeners = new ArrayList<>();
```

Comment:
This method is called individually for every single node of the cluster. So why do we need to separate the `brokerAdvertisedListeners` and `controllerAdvertisedListeners`? It would always end up in the same `advertised.listeners` option, or? So why not keep just `advertisedListeners`?

Reply:
I think it was just little more readable for me through all the if conditions but it's actually wasn't necessary. I changed it back to just advertisedListeners.